### PR TITLE
Remove some deprecated code

### PR DESCRIPTION
**Pull request by Laerte**
Closes: #6100
Closes: #6101
Closes: #6102
Closes: #6103
Closes: #6104
Closes: #6105
Closes: #6106
Closes: #6107
Closes: #6110

### CODE REVIEW

File: scrapy/pipelines/media.py [hunk 1]
```diff
@@ -126,37 +123,6 @@ def _process_request(self, request, info, item):
...
-            "image_downloaded",
-            "get_images",
-        ]
-
-        for method_name in methods:
-            method = getattr(self, method_name, None)
-            if callable(method):
-                setattr(self, method_name, self._compatible(method))
-
-    def _compatible(self, func):
```
Comment (Laerte) [inline scrapy/pipelines/media.py:146]: Removed as well because is not called anymore after  `_make_compatible` removal.

Comment (wRAR) [reply scrapy/pipelines/media.py:146]: Yeah, this was the intention.

File: tests/test_downloadermiddleware_httpcompression.py [hunk 2]
```diff
@@ -381,20 +380,6 @@ def __init__(self):
                 super().__init__()
 
         crawler = get_crawler(Spider)
-        with catch_warnings(record=True) as caught_warnings:
+
+        with self.assertRaises(TypeError):
```
Comment (wRAR) [inline tests/test_downloadermiddleware_httpcompression.py:384]: Maybe we should just delete this test?

File: tests/test_feedexport.py [hunk 3]
```diff
@@ -2850,20 +2850,8 @@ def test_init(self):
...
-            feed_exporter = FeedExporter.from_crawler(crawler)
-
-        spider = scrapy.Spider("default")
-        with pytest.warns(
-            ScrapyDeprecationWarning,
-            match="StdoutFeedStorageWithoutFeedOptions does not support "
-            "the 'feed_options' keyword argument.",
-        ):
-            feed_exporter.open_spider(spider)
+        with pytest.raises(TypeError):
```
Comment (wRAR) [inline tests/test_feedexport.py:2853]: Similarly, I think all these `*PreFeedOptionsTest` tests can be removed.

Comment (Laerte) [reply tests/test_feedexport.py]: Done.

**Review by wRAR — APPROVED** [review]
Thanks!

**Review by Gallaecio — APPROVED** [review]
Awesome. I think this is the most issues a single PR has fixed at once :slightly_smiling_face:

**Pull request merged**